### PR TITLE
Tag suggest

### PR DESCRIPTION
**Pull request by jessemtso**
this should fix a lot of the tag suggestion issues.

also sorry damien, i forgot to squash them into 1 commit, but its a clean branch this time :D

### CODE REVIEW

File: dao.py [hunk 1]
```diff
@@ -267,13 +267,30 @@ def get_tournament_by_id(self, id):
...
+            re.split(re_test_1, alias_lower)[2],
+            re.split(re_test_2, alias_lower)[2],
+            # well, we're using set, so why not
+            re.split(re_test_1, alias_lower)[2].strip(),
+            re.split(re_test_2, alias_lower)[2].strip(),
         ]))
+
+        # XXX add in all the components, sort of a last ditch hack. possibly take this out if its getting too many false positives
+        for s in re.split(" ", alias_lower):
+            if s not in similiar_aliases:
```
Comment (jiangtyd) [inline dao.py]: it's a set, so you can just do `similar_aliases.update(re.split(" ", alias_lower))`

on the other hand, I don't think we should just add everything in `alias_lower.split()`; I think we should only include suffixes, i.e.
```python
alias_words = alias_lower.split()
similar_aliases.update([' '.join(alias_words[i:]) for i in xrange(len(alias_words))])
```

**Comment by jiangtyd** [comment]
i'd prefer a squash before you merge

@yedi github has a new repo option to add a "squash and merge" button, can you enable that?

File: dao.py [hunk 2]
```diff
@@ -267,14 +267,29 @@ def get_tournament_by_id(self, id):
...
+            re.split(special_chars, alias_lower)[-1].strip(),
+            # regex nonsense to deal with pool prefixes
+            re.split(re_test_1, alias_lower)[2],
+            re.split(re_test_2, alias_lower)[2],
+            # well, we're using set, so why not
+            re.split(re_test_1, alias_lower)[2].strip(),
+            re.split(re_test_2, alias_lower)[2].strip(),
+        ])
+
+        alias_words = alias_lower.split()
```
Comment (jiangtyd) [inline dao.py]: comment here saying "add suffixes of string"

File: test/test_dao.py [hunk 3]
```diff
@@ -777,3 +777,11 @@ def test_get_and_insert_pending_merge(self):
...
         dao = self.norcal_dao
         self.assertIsNone(dao.get_pending_merge(ObjectId("420f53650181b84aaaa01051"))) #mlg1337noscope
+
+    def test_get_players_with_similar_alias(self):
+        dao = self.norcal_dao
+        self.assertTrue("gaR" in dao.get_players_with_similar_alias("1 1 gar"))
+        self.assertTrue("gaR" in dao.get_players_with_similar_alias("p1s1 gar"))
+        self.assertTrue("gaR" in dao.get_players_with_similar_alias("GOOG| gar"))
+        self.assertTrue("gaR" in dao.get_players_with_similar_alias("GOOG | gar"))
+        self.assertTrue("gaR" in dao.get_players_with_similar_alias("p1s2 GOOG| gar"))
```
Comment (jiangtyd) [inline test/test_dao.py]: add another test for `"garpr gar"`

**Comment by jiangtyd** [comment]
k two more small comments then :+1:

**Comment by jessemtso** [comment]
tests are passing, merging!

**Pull request merged**